src/input/keyboard.c

```diff
@@ -386,11 +387,34 @@ handle_key_release(struct server *server, uint32_t evdev_keycode)
 		end_cycling(server);
 	}
 
+	key_state_bound_key_remove(evdev_keycode);
+
+	/*
+	 * There are some clients (for example blender) that want to see the
+	 * modifier-release-event even if it was part of a keybinds. This is
+	 * treated as a special case and can only be achieved by configuration.
+	 *
+	 * Most clients (including those using Qt and GTK) are setup to not see
+	 * these modifier release events - and actually misbehave if they do.
+	 * For example Firefox shows the menu bar if alt is pressed and then
+	 * released, whereas if only pressed (because the release is absorbed)
+	 * nothing happens. So, if Firefox saw bound modifier-release-events it
+	 * would show the menu bar every time the window-switcher is used with
+	 * alt-tab.
+	 */
+	struct view *view = server->active_view;
+	if (view && window_rules_get_property(view,
+			"wantAbsorbedModifierReleaseEvents") == LAB_PROP_TRUE) {
+		struct wlr_keyboard *kb = &server->seat.keyboard_group->keyboard;
+		if (is_modifier(kb, evdev_keycode)) {
+			return false;
+		}
+	}
```

Comment:
Could we turn this around, e.g. check `is_modifier()` first? As this is a somewhat hot path and called on every key release I'd prefer if we could reduce the overhead as much as possible. At some point we should also convert the string properties to enums similar to what we do with actions.

Reply:
We can pass `struct keyinfo` to this function rather than just the keycode. It contains a `.is_modifier` bool.

Reply:
Okay - will do. 

Just to note though - it's only for release events of keys that were bound.

Reply:
> Just to note though - it's only for release events of keys that were bound.

Right, missed the early out in `handle_key_release()`.
She the other branch in `handle_compositor_keybindings()` with `cur_keybind && cur_keybind->on_release` also return false in this case?

Reply:
^ No.

`handled==false` means that the key-event should be passed onto the focused client.

`cur_keybind->on_release` means that we have bound an action to the release event (like super to launch menu), and in this case we do NOT want to pass on the event to the client.